internal/command/sandbox/delete.go

```diff
@@ -79,7 +82,53 @@ func delete(cfg *config.SandboxDelete, out io.Writer, args []string) error {
 		return err
 	}
 
-	fmt.Fprintf(out, "Deleted sandbox %q.\n", name)
+	fmt.Fprintf(out, "Deleted sandbox %q.\n\n", name)
 
+	if cfg.Wait {
+		// Wait for the API server to completely reflect deletion.
+		if err := waitForDeleted(cfg, out, id); err != nil {
+			fmt.Fprintf(out, "\nDeletion was initiated, but the sandbox may still exist in a terminating state. To check status, run:\n\n")
+			fmt.Fprintf(out, "  signadot sandbox get-status %v\n\n", name)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func waitForDeleted(cfg *config.SandboxDelete, out io.Writer, sandboxID string) error {
+	fmt.Fprintf(out, "Waiting (up to --wait-timeout=%v) for sandbox to finish terminating...\n", cfg.WaitTimeout)
+
+	params := sandboxes.NewGetSandboxStatusByIDParams().WithOrgName(cfg.Org).WithSandboxID(sandboxID)
+
+	spin := spinner.Start(out, "Sandbox status")
+	defer spin.Stop()
+
+	err := poll.Until(cfg.WaitTimeout, func() bool {
+		result, err := cfg.Client.Sandboxes.GetSandboxStatusByID(params, nil)
+		if err != nil {
+			// If it's a "not found" error, that's what we wanted.
+			// TODO: Pass through an error code so we don't have to rely on the error message.
+			if strings.Contains(err.Error(), "can't get sandbox status: not found") {
+				spin.StopMessage("Terminated")
+				return true
+			}
+
+			// Otherwise, keep retrying in case it's a transient error.
+			spin.Messagef("error: %v", err)
+			return false
+		}
+		status := result.Payload.Status
+		if status.Ready {
+			spin.Message("Waiting for sandbox to begin terminating.")
```

Comment:
seems like a weird error message - maybe we group this under "Waiting for sandbox to terminate"?

Reply:
Done.